sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -190,7 +190,8 @@ abstract class Optimizer(catalogManager: CatalogManager)
       ReplaceDeduplicateWithAggregate) ::
     Batch("Aggregate", fixedPoint,
       RemoveLiteralFromGroupExpressions,
-      RemoveRepetitionFromGroupExpressions) :: Nil ++
+      RemoveRepetitionFromGroupExpressions,
+      EliminateUnnecessaryOuterJoin) :: Nil ++
```

Comment:
Maybe, `RemoveUnnecessaryOuterJoin` looks similar with the others here? And, shorter?

Reply:
ok